components/uaepass-authenticator/src/main/java/org/wso2/carbon/identity/uaepass/authenticator/UAEPassAuthenticator.java

```diff
@@ -0,0 +1,1109 @@
+            String tokenEndPoint = getTokenUrl(envUAEPass);
+            String callbackUrl = authenticatorProperties.get(UAEPassAuthenticatorConstants.UAE.CALLBACK_URL);
+
+            accessTokenRequest = OAuthClientRequest
+                    .tokenLocation(tokenEndPoint)
+                    .setGrantType(GrantType.AUTHORIZATION_CODE)
+                    .setClientId(clientId).setClientSecret(clientSecret)
+                    .setRedirectURI(callbackUrl)
+                    .setCode(authzResponse.getCode())
+                    .buildBodyMessage();
+
+            if (accessTokenRequest != null) {
+                String serverURL = ServiceURLBuilder.create().build().getAbsolutePublicURL();
+                accessTokenRequest.addHeader(UAEPassAuthenticatorConstants.UAE.HTTP_ORIGIN_HEADER, serverURL);
+            }
+
+        } catch (OAuthSystemException e) {
+            LOG.error("Unable to build the request with request's body attributes.");
+            throw new UAEPassAuthnFailedException("Exception while building access token request " +
+                    "with the request body", e);
+        } catch (URLBuilderException e) {
+            if(LOG.isDebugEnabled()) {
+                LOG.error("Unable to identify common-auth URL on browser");
+            }
+            throw new UAEPassAuthnFailedException("Error occurred while extracting the absolute public URL from browser.", e);
+        }
+        return accessTokenRequest;
+    }
+
+
+    /**
+     * Returns the OAuth type response to the back channel.
+     *
+     * @param oAuthClient
+     * @param accessRequest
+     * @return OAuthClientResponse
+     * @throws AuthenticationFailedException
+     */
+    public OAuthClientResponse getOauthResponse(OAuthClient oAuthClient, OAuthClientRequest accessRequest)
+            throws OAuthSystemException, OAuthProblemException {
+
+        OAuthClientResponse oAuthResponse;
+        oAuthResponse = oAuthClient.accessToken(accessRequest);
+        return oAuthResponse;
+    }
+
+
+    /**
+     * This method specifies the OIDC to the state parameter.
+     *
+     * @param request
+     * @return String
+     */
+    public String getLoginType(HttpServletRequest request) {
+
+        String state = request.getParameter(UAEPassAuthenticatorConstants.UAE.OAUTH2_PARAM_STATE);
+        if (StringUtils.isNotBlank(state)) {
+                String[] stateElements = state.split(",");
+                if (stateElements.length > 1) {
+                    return stateElements[1];
+                }
+        }
+        if(LOG.isDebugEnabled()){
+            LOG.error("Empty split elements in state");
+        }
+        return null;
+    }
+
+
+    /**
+     * This method is used to modify commonAuth logout URL, where ever relevant to WSO2 Identity Server.
+     *
+     * @param logOutUri
+     * @return
+     */
+    public String modifyLogoutUrl(String logOutUri) {
```

Comment:
This type of method can not be considered utils. Since those are used regularly by the authenticator. Even for other outbound authenticators, those methods were defined as the same. In such a case do we need to extract those?